commit 6ed523649e296a16494c6b559ef22de04833cddc
Author: hasufell <hasufell@posteo.de>
Date:   Mon Sep 23 23:57:48 2013 +0200

    BUILD: small cleanup to systemd bits
    
    * don't double check for pkg-config
    * use AS_IF
    * consistent variable naming
    * do not install systemd file when "--with-systemd" is omitted

diff --git a/pgl/Makefile.am b/pgl/Makefile.am
index a81c3ee..c5dd377 100644
--- a/pgl/Makefile.am
+++ b/pgl/Makefile.am
@@ -6,7 +6,7 @@ ACLOCAL_AMFLAGS = -I m4
 
 # Ensure that make distcheck continues to work
 DISTCHECK_CONFIGURE_FLAGS = \
-	--with-systemdsystemunitdir=$$dc_install_base/$(systemdsystemunitdir)
+	--with-systemd=$$dc_install_base/$(SYSTEMDUNITDIR)
 
 SUBDIRS = \
 	docs \
diff --git a/pgl/configure.ac b/pgl/configure.ac
index 63b443c..dd2086e 100644
--- a/pgl/configure.ac
+++ b/pgl/configure.ac
@@ -14,9 +14,7 @@
 AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_LIBTOOL
-AC_PATH_PROG([PKGCONFIG], [pkg-config])
-AS_IF([test "x$PKGCONFIG" = "x"],
-    [AC_MSG_ERROR([pkg-config not found!])])
+PKG_PROG_PKG_CONFIG
 
 # Checks for header files.
 AC_CHECK_HEADERS([arpa/inet.h inttypes.h limits.h netinet/in.h stdlib.h string.h sys/time.h syslog.h unistd.h])
@@ -134,14 +132,14 @@
 
 # use systemd service file (yes, if user specifies a path)
 # http://www.freedesktop.org/software/systemd/man/daemon.html
-PKG_PROG_PKG_CONFIG
-AC_ARG_WITH([systemdsystemunitdir],
-        AS_HELP_STRING([--with-systemdsystemunitdir=DIR], [Directory for systemd service files]),
-        [], [with_systemdsystemunitdir=$($PKG_CONFIG --variable=systemdsystemunitdir systemd)])
-if test "x$with_systemdsystemunitdir" != xno; then
-        AC_SUBST([systemdsystemunitdir], [$with_systemdsystemunitdir])
-fi
-AM_CONDITIONAL(HAVE_SYSTEMD, [test -n "$with_systemdsystemunitdir" -a "x$with_systemdsystemunitdir" != xno ])
+AC_ARG_WITH([systemd],
+        [AS_HELP_STRING([--with-systemd=UNITDIR], [install systemd unit file (into UNITDIR if given)])],
+        [with_systemd="$withval"],
+        [with_systemd="no"])
+AS_IF([test "x$with_systemd" != xno],
+	  [AS_IF([test "$with_systemd" = yes], [with_systemd="`$PKG_CONFIG --variable=systemdsystemunitdir systemd`"])]
+	  [AC_SUBST([SYSTEMDUNITDIR], [$with_systemd])])
+AM_CONDITIONAL(HAVE_SYSTEMD, [test -n "$with_systemd" -a "x$with_systemd" != xno ])
 
 # let user specify iconsdir
 AC_ARG_WITH([iconsdir],
@@ -321,9 +319,9 @@
 else
 echo QT-gui....................................... : no
 fi
-
-if test -n "$with_systemdsystemunitdir" -a "x$with_systemdsystemunitdir" != xno; then
-echo systemdsystemunitdir......................... : $systemdsystemunitdir
+if test -n "$with_systemd" -a "x$with_systemd" != xno; then
+echo systemd ..................................... : yes
+echo systemd unit dir ............................ : $with_systemd
 else
 echo systemd...................................... : no
 fi
diff --git a/pgl/pglcmd/Makefile.am b/pgl/pglcmd/Makefile.am
index c7f34a5..d06871e 100644
--- a/pgl/pglcmd/Makefile.am
+++ b/pgl/pglcmd/Makefile.am
@@ -51,6 +51,7 @@ pgllib_DATA = \
 if HAVE_SYSTEMD
 systemdsystemunit_DATA = \
 	init/pgl.service
+systemdsystemunitdir = @SYSTEMDUNITDIR@
 endif
 
 # Don't update PATH here anymore, because on user's make it doesn't contain
